### PR TITLE
Allows pAI's to print photographs for 10 ram

### PR DESCRIPTION
**Pull request by V2LenKagamine**
Also moves the "Internal GPS' down by its 35 ram brother.

## About The Pull Request

Allows PAI to access a newly inbuilt bluespace printer at the cost of some processing power being diverted to mine for cryptocurrency.
## Why It's Good For The Game
Pai's are perfect travel buddies, but its a shame they cant print out their memories to share with their master. Well now they can! Simply take a picture, select the "Printer" button, and (optionally select an image if multiple are available) Presto! Instant picture. No toner required, its ported directly from an automatic nanotransen printer! Of course, some ram is actually being used for crypto mining for pure NT profit, but who cares? Pictures!

## Changelog

:cl:
add: Pai Printer Module
/:cl:

### CODE REVIEW

File: code/modules/photography/camera/silicon_camera.dm [hunk 1]
```diff
@@ -89,3 +89,13 @@
 	C.toner -= printcost  //All fun allowed.
 	visible_message(span_notice("[C.name] spits out a photograph from a narrow slot on its chassis."))
 	to_chat(usr, span_notice("You print a photograph."))
+
+/obj/item/camera/siliconcam/proc/paiprint(mob/user)
+	var/mob/living/silicon/pai/C = loc
```
Comment (Fikou) [inline code/modules/photography/camera/silicon_camera.dm:94]: the fuck does C mean

Comment (V2LenKagamine) [reply code/modules/photography/camera/silicon_camera.dm:94]: C is just generic variable used for the "visable message". Basically originally was a copy-pasta of borgoprint but I retooled it to use the default camera function instead.

Comment (MrDoomBringer) [reply code/modules/photography/camera/silicon_camera.dm:94]: https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#dont-use-abbreviations

Comment (V2LenKagamine) [reply code/modules/photography/camera/silicon_camera.dm]: fixed.

File: code/modules/photography/camera/silicon_camera.dm [hunk 2]
```diff
@@ -89,3 +89,13 @@
...
 	to_chat(usr, span_notice("You print a photograph."))
+
+/obj/item/camera/siliconcam/proc/paiprint(mob/user)
+	var/mob/living/silicon/pai/paimob = loc
+	var/datum/picture/selection = selectpicture(user)
+	if(!istype(selection))
+		to_chat(user, span_warning("Invalid Image."))
+		return
+	printpicture(user,selection)
+	visible_message(span_notice("A picture appears on top of the chassis of [paimob.name]!"))
```
Comment (Cyberboss) [inline code/modules/photography/camera/silicon_camera.dm:100]: Missing a loc for this visible_message

File: code/modules/photography/camera/silicon_camera.dm [hunk 3]
```diff
@@ -89,3 +89,13 @@
...
+	to_chat(usr, span_notice("You print a photograph."))
```
Comment (Cyberboss) [inline code/modules/photography/camera/silicon_camera.dm:101]:
```suggestion
	to_chat(user, span_notice("You print a photograph."))
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/photography/camera/silicon_camera.dm (hunk 2 above)
Comment (V2LenKagamine) [reply code/modules/photography/camera/silicon_camera.dm:100]: paimob = loc

if this isnt what you mean explain in DIsc.

Comment (Cyberboss) [reply code/modules/photography/camera/silicon_camera.dm:100]: Sorry, this just looked off. The reason is it's missing a "self" version of the message. Check the definition of `proc/visible_message`. You should add that and then remove the `to_chat` call below.

Comment (V2LenKagamine) [reply code/modules/photography/camera/silicon_camera.dm]: should have fixed it!

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**